spec/models/authenticator/ldap_spec.rb

```diff
@@ -155,6 +156,23 @@ def normalize(dn)
     end
   end
 
+  describe '#authorize_queue' do
+    def authenticate
+      subject.authenticate(username, password)
+    end
+
+    let(:username) { 'alice' }
+    let(:password) { 'secret' }
+    let(:config) { super().merge(:ldap_role => true) }
+
+    it "encrypts password for queuing" do
+      allow(MiqTask).to receive(:create).and_return(double(:id => :return_value))
+      allow(MiqQueue).to receive(:put) { :return_value }
+      authenticate
+      expect(config[:bind_pwd]).to match(/v([0-9]+):\{([^}]*)\}/)
+    end
+  end
```

Comment:
Yeah, I agree, although I might go further.  In this test, I don't care how `encrypt_ldap_password`  or `decrypt_ldap_password ` implements it, only that they get called if we're ldap.  I'd rather not pull in `MiqPassword` as a dependency for this high level test as we're be more tied to it.  I might just expect decrypt_ldap_password or encrypt is received.

In the test for the individual methods, there we might want to expect MiqPassword as @chrisarcand suggests.